src/server/pfs/server/driver_file.go

```diff
@@ -450,3 +447,49 @@ func (d *driver) getFileset(pachClient *client.APIClient, commit *pfs.Commit) (*
 	ids = append(ids, *id)
 	return d.storage.Compose(pachClient.Ctx(), ids, defaultTTL)
 }
+
+func (d *driver) getOrComputeTotal(pachClient *client.APIClient, commit *pfs.Commit) (*fileset.ID, error) {
+	ctx := pachClient.Ctx()
+	commitInfo, err := d.getCommit(pachClient, commit)
+	if err != nil {
+		return nil, err
+	}
+	if commitInfo.Finished == nil {
+		return nil, errors.Errorf("attempted to compute total of unfinished commit")
+	}
+	commit = commitInfo.Commit
+	id, err := d.commitStore.GetTotalFileset(ctx, commit)
```

Comment:
I would prefer this to return an error that indicates it doesn't exist, then we check for that to determine if we need to compute it.

Reply:
I was thinking about that, but returning the not exist error from the fileset package is ambiguous, so we would need an error type specifically for this case.  And it would just complicate the error check every time it is used.  No fileset is also a valid result, as opposed to in the fileset package where asking for a fileset that doesn't exist is always an error.

Reply:
I don't think it is really that ambiguous. If the total fileset doesn't exist, then it doesn't exist. You need a check in both cases, either for a certain type of error or whether the ID is nil. The certain type of error approach is going to be much less prone to bugs, than returning a nil ID and nil error.